tests/Api.IntegrationTests/Helpers/WebDriverFactory.cs

```diff
@@ -1,13 +1,42 @@
+using System.Diagnostics;
 using OpenQA.Selenium;
 using OpenQA.Selenium.Chrome;
 using OpenQA.Selenium.Internal;
+using OpenQA.Selenium.Support.UI;
 using OpenQA.Selenium.VirtualAuth;
 
 namespace Passwordless.Api.IntegrationTests.Helpers;
 
 public static class WebDriverFactory
 {
     public static WebDriver GetWebDriver(string driverUrl)
+    {
+        for (int i = 0; i < 5; i++)
+        {
+            ChromeDriver? driver = null;
+            try
+            {
+                (driver, var res) = GetDriver(driverUrl);
+                if (res)
+                {
+                    Console.WriteLine("Driverdebug: Successfully created driver: " + i);
+                    return driver;
+                }
+            }
+            catch (WebDriverTimeoutException e)
+            {
+                Console.WriteLine(e);
+            }
+            Console.WriteLine("Driverdebug: Waited but need to retry: " + i);
+
+            driver?.Quit();
+
```

Comment:
@abergs  Calling `driver?.Quit()` would shut down the browser, only having to wait again and take up resources.